test/test_enrolments_controller.py

```diff
@@ -0,0 +1,138 @@
+from test.party_client import PartyTestClient
+
+from sqlalchemy.exc import DataError
+from sqlalchemy.orm.exc import NoResultFound
+
+from ras_party.controllers.enrolments_controller import enrolments_by_parameters
+from ras_party.models.models import (
+    Business,
+    BusinessRespondent,
+    Enrolment,
+    EnrolmentStatus,
+    Respondent,
+)
+from ras_party.support.session_decorator import with_db_session
+
+respondents_enrolments = [
+    {
+        "respondent": "b6f9d6e8-b840-4c95-a6ce-9ef145dd1f85",
+        "enrolment_details": [
+            {
+                "business": "75d9af56-1225-4d43-b41d-1199f5f89daa",
+                "survey_id": "9200d295-9d6e-41fe-b541-747ae67a279f",
+                "status": EnrolmentStatus.ENABLED,
+            },
+            {
+                "business": "98e2c9dd-a760-47dd-ba18-439fd5fb93a3",
+                "survey_id": "c641f6ad-a5eb-4d82-a647-7cd586549bbc",
+                "status": EnrolmentStatus.ENABLED,
+            },
+        ],
+    },
+    {
+        "respondent": "5718649e-30bf-4c25-a2c0-aaa733e54ed6",
+        "enrolment_details": [
+            {
+                "business": "af25c9d5-6893-4342-9d24-4b88509e965f",
+                "survey_id": "9200d295-9d6e-41fe-b541-747ae67a279f",
+                "status": EnrolmentStatus.ENABLED,
+            },
+            {
+                "business": "75d9af56-1225-4d43-b41d-1199f5f89daa",
+                "survey_id": "9200d295-9d6e-41fe-b541-747ae67a279f",
+                "status": EnrolmentStatus.DISABLED,
+            },
+        ],
+    },
+]
+
+
+class TestEnrolments(PartyTestClient):
```

Comment:
There are many different other combinations I could have done, but I don't think they are need, so covered each many one individually, a combination and then the errors that could be raised